src/pydsl/triton.py

```diff
@@ -0,0 +1,237 @@
+        sign_val = "i" if pytype.sign == Sign.UNSIGNED else "i"
+        return sign_val + str(pytype.width)
+    if issubclass(pytype, Float):
+        return "fp" + str(pytype.width)
+    # Handle constexpr (for BLOCK_SIZE, etc.)
+    if issubclass(pytype, Number):
+        return "constexpr"
+    raise TypeError(
+        f"converting {pytype.__qualname__} to a Triton type is not supported"
+    )
+
+
+def remove_unnecessary_args(func_op: func.FuncOp) -> None:
+    """
+    Removes unnecessary arguments from the funcOp.
+    This is used to clean up the MLIR after triton-adapter-opt.
+    """
+    # Find the function in the module
+    entry_block = func_op.entry_block
+    if entry_block is None:
+        raise AssertionError(
+            f"Function {func_op.sym_name.value} has no entry block."
+        )
+
+    # Collect indices of unused arguments
+    indices_to_remove = []
+    for i, arg in enumerate(entry_block.arguments):
+        # `arg.uses` is an iterable of OperationUse;
+        # if it's empty, the arg is unused. We also check if it's the first
+        # or last 6 arguments, since these are the ones added by triton-adapter-opt
+        # and we want to keep unused arguments that are user-defined.
+        if not list(arg.uses) and (
+            i == 0 or i >= len(entry_block.arguments) - 6
+        ):
+            indices_to_remove.append(i)
+
+    if not indices_to_remove:
+        return  # No unused arguments
+
+    indices_to_remove.reverse()
+
+    # Update argument attributes if present
+    if "arg_attrs" in func_op.attributes:
+        arg_attrs = func_op.arg_attrs
+        new_attrs = [
+            attr
+            for i, attr in enumerate(arg_attrs)
+            if i not in indices_to_remove
+        ]
+        func_op.arg_attrs = mlirir.ArrayAttr.get(new_attrs)
+
+    # Remove unused arguments from the entry block
+    for idx in indices_to_remove:
+        entry_block.erase_argument(idx)
+
+    # Update function type
+    old_func_type = func_op.type
+    new_inputs = [
+        typ
+        for i, typ in enumerate(old_func_type.inputs)
+        if i not in indices_to_remove
+    ]
+    new_func_type = mlirir.FunctionType.get(new_inputs, old_func_type.results)
+    func_op.attributes["function_type"] = mlirir.TypeAttr.get(new_func_type)
+
+
+def combine_mlir_modules(
+    main_module: mlirir.Module,
+    module2: mlirir.Module,
+    triton_funcs: dict,
+    current_signature: tuple[str, ...],
+) -> func.FuncOp:
+    """
+    Combines two MLIR modules by copying the operations from `module2` into
+    `main_module`. checks to see if the function has already been created, and
+    if it has, checks to see if the signature has been created. If the function
+    exists with a different signature, add "n" to the function name, where "n"
+    is the current number of distinct signatures.
+    """
+    with InsertionPoint(main_module.body):
+        prefix = "__triton_to_pydsl_"
+        callee = []
```

Comment:
```suggestion
        callees = []
```
This makes a bit more sense with this code structure (even if we later assert it has length 1), right?